src/tracer.c

```diff
@@ -195,42 +147,89 @@ handle_syscall(pid_t pid, const struct ptrace_syscall_info *entry,
 	return;			       // return on syscall failure
     }
 
-    const int syscall = entry->entry.nr;
-
-    if (syscall != SYS_open && syscall != SYS_creat && syscall != SYS_openat
-	&& syscall != SYS_close) {
-	return;			       // return if we don't care about
-	// tracking said syscall
-    }
-
-    if (syscall == SYS_close) {
-	handle_close(pinfo->finfo + entry->entry.args[0]);
-	return;
-    }
-
-    const int fd = exit->exit.rval;
+    int fd;
+    void *path;
+    int flags;
+    int dirfd;
 
-    PROCESS_INFO *pinfo = find(pid);
-    FILE_INFO *finfo;
+    switch (entry->entry.nr) {
+	case SYS_open:
+	    // int open(const char *pathname, int flags, ...);
+	    fd = (int) exit->exit.rval;
+	    path = (void *) entry->entry.args[0];
+	    flags = (int) entry->entry.args[1];
 
-    if (fd == pinfo->finfo_size) {
-	finfo = next_finfo(pinfo);
-    } else {
-	finfo = pinfo->finfo + fd;
-    }
+	    FILE_INFO *finfo = finfo_at(find(pid), fd);
```

Comment:
have the definition of `FILE_INFO *finfo` together with the ones above (`fd`, etc.)

Reply:
@zvr I'd rather not do that, we have more sys handlers to add and not every one shares this part in order to extract it out of the `switch`. Do you still wish to?